illumos-utils/src/vmm_reservoir.rs

```diff
@@ -0,0 +1,39 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Wrappers around reservoir controls
+
+use omicron_common::api::external::ByteCount;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Reservoir size must be a multiple of MiB, not: {0}")]
+    InvalidSize(ByteCount),
+
+    #[error("Failed to set reservoir size: {0}")]
+    ReservoirError(#[from] std::io::Error),
+}
+
+/// Controls the size of the memory reservoir
+pub struct ReservoirControl {}
+
+impl ReservoirControl {
+    /// Sets the reservoir to a particular size.
+    ///
+    /// - "size" must be exactly divisible by a MiB
```

Comment:
Should this just take the size in `MiB` then?

Reply:
We *could*, but would probably want a new type to represent that, since IMO it's confusing to have `u64`s with overloaded meaning. This was why we originally used `ByteCount` everywhere, to make it explicit that "it's a count of bytes", but that does have the downside of permitting values that need to be validated.

Do you have a strong opinion on where this new type should live?